src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py

```diff
@@ -87,7 +116,34 @@ def get_postgresql_management_client(cli_ctx, **_):
                 secret=getenv(CLIENT_SECRET),
                 tenant=getenv(TENANT_ID))
         else:
-            from msrest.authentication import Authentication    # pylint: disable=import-error
+            from msrest.authentication import Authentication  # pylint: disable=import-error
+            credentials = Authentication()
+
+        return PostgreSQLManagementClient(
+            subscription_id=getenv(SUB_ID_OVERRIDE),
+            base_url=rm_uri_override,
+            credentials=credentials)
+    # Normal production scenario.
+    return get_mgmt_service_client(cli_ctx, PostgreSQLManagementClient)
```

Comment:
Is there anything special for Authentication() and credentials?
In addition, `credentials` seems  a track2 SDK property.

Reply:
If there is a client_id we pass service principal creds; this is mainly for the Python SDK & follows what we setup previously for past releases of rdbms.